### PR TITLE
Manifest i18n review

### PR DESCRIPTION
**Pull request by zhangyongjing**
Addressing i18n comments with the following changes:
1. relax `appID` string format constraints
2. relax `appName` string format constraints
3. clarify that `dir`, `lang` applies to all "localizable attributes"
4. clarify widget.path format

@aphillips @ShourenLan @xfq @zhiqiangyu

### CODE REVIEW

**Comment by xfq** [comment]
* Preview: https://raw.githack.com/zhangyongjing/miniapp/manifest-i18n-review/specs/manifest/index.html
* Diff: https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Fminiapp%2Fspecs%2Fmanifest%2F&doc2=https%3A%2F%2Fraw.githack.com%2Fzhangyongjing%2Fminiapp%2Fmanifest-i18n-review%2Fspecs%2Fmanifest%2Findex.html

File: specs/manifest/index.html [hunk 1]
```diff
@@ -308,10 +308,10 @@ <h2>
...
+          <code>dir</code> specifies the default base direction of the text value of the whole MiniApp as well as the <a>localizable attributes</a> in the manifest file such as the <code>appName</code>, <code>shortName</code> and <code>description</code> attributes. The value of <code>dir</code> can be <code>ltr</code> or <code>rtl</code>. <code>ltr</code> means left-to-right, <code>rtl</code> means right-to-left. The default value is <code>ltr</code>.
         </p>
         <p its-locale-filter-list="zh-hans" lang="zh-hans">
-          dir，指定名称、简称和描述成员的主文本方向。这里取值类型可分为：<code>ltr</code>和<code>rtl</code>，<code>ltr</code>表示从左到右，<code>rtl</code>表示从右到左。默认值：<code>ltr</code>。
+          <code>dir</code>定义了整个MiniApp以及Manifest文件中<code>appName</code>, <code>shortName</code> and <code>description</code>等<a>可本地化属性</a>的默认主文本方向，取值类型可以为：<code>ltr</code>和<code>rtl</code>，<code>ltr</code>表示从左到右，<code>rtl</code>表示从右到左。默认值为<code>ltr</code>。
```
Comment (xfq) [inline specs/manifest/index.html:314]:
```suggestion
          <code>dir</code>定义了整个MiniApp以及Manifest文件中<code>appName</code>、<code>shortName</code>和<code>description</code>等<a>可本地化属性</a>的默认主文本方向，取值类型可以为：<code>ltr</code>和<code>rtl</code>，<code>ltr</code>表示从左到右，<code>rtl</code>表示从右到左。默认值为<code>ltr</code>。
```

File: specs/manifest/index.html [hunk 2]
```diff
@@ -321,11 +321,10 @@ <h2>
           <span its-locale-filter-list="zh-hans" lang="zh-hans">语言</span>
         </h2>
         <p its-locale-filter-list="en" lang="en">
-            <code>lang</code> specifies the language tag of MiniApp. The value here takes the form of
-            <code>Language-Tag</code> as defined in the [[BCP47]]</a>. The default value is <code>zh-Hans</code>. If not specified, the default value is used.
+            <code>lang</code> specifies the default language of the MiniApp. It's applied to all <a>localizable attributes</a> of the manifest. The value here MUST be a well-formed <code>Language-Tag</code> specified in [[BCP47]].  The default value is undefined. Implementers are encouraged to provide the accurate value that matches the manifest content.
         </p>
         <p its-locale-filter-list="zh-hans" lang="zh-hans">
-            <code>lang</code>指定MiniApp的语言标签。这里取值为<code>Language-Tag</code>值，详见[[BCP47]]</a>。默认值是<code>zh-Hans</code>。未指定的情况下，将采用默认值。
+            <code>lang</code>定义了MiniApp的默认语言，适用于manifest所有<a>可本地化属性</a>，取值必须为[[BCP47]]所规定的合法的<code>Language-Tag</code>值。默认值未定义。开发者需要提供匹配manifest内容的准确值。
```
Comment (xfq) [inline specs/manifest/index.html:327]:
```suggestion
            <code>lang</code>定义了MiniApp的默认语言，适用于manifest所有<a>可本地化属性</a>，取值必须为[[BCP47]]所规定的合法的（well-formed）<code>Language-Tag</code>值。默认值未定义。开发者需要提供匹配manifest内容的准确值。
```

Comment (xfq) [reply specs/manifest/index.html:327]: 这里建议加上英文原文，因为「well-formed」是一个BCP 47的术语，和valid的意思不同。

File: specs/manifest/index.html [hunk 3]
```diff
@@ -693,6 +692,14 @@ <h2>
             </tr>
           </tbody>
         </table>
+
+        <p its-locale-filter-list="en" lang="en">
+            <code>window</code> inherits the text direction and language configuration from <code>dir</code> and <code>lang</code>.
+          </p>
+          <p its-locale-filter-list="zh-hans" lang="zh-hans">
+            <code>window</code>的文字方向和语言设置继承自<code>dir</code> and <code>lang</code>.
```
Comment (xfq) [inline specs/manifest/index.html:700]:
```suggestion
            <code>window</code>的文字方向和语言设置继承自<code>dir</code>和<code>lang</code>.
```

File: specs/manifest/index.html [hunk 4]
```diff
@@ -749,8 +756,8 @@ <h2>
                 <span its-locale-filter-list="zh-hans" lang="zh-hans">是</span>
               </td>
               <td>
-                <span its-locale-filter-list="en" lang="en">Corresponding page path of Widget</span>
-                <span its-locale-filter-list="zh-hans" lang="zh-hans">Widget对应页面路径</span>
+                <span its-locale-filter-list="en" lang="en">Corresponding page path of Widget.It follows the same path route format as defined in <code>pages</code>.</span>
```
Comment (xfq) [inline specs/manifest/index.html:759]:
```suggestion
                <span its-locale-filter-list="en" lang="en">Corresponding page path of Widget. It follows the same path route format as defined in <code>pages</code>.</span>
```

File: specs/manifest/index.html [hunk 5]
```diff
@@ -846,6 +853,29 @@ <h2>
...
+    <section>
+        <h2>
+            <span its-locale-filter-list="en" lang="en"><dfn>Localizable attributes</dfn></span>
+            <span its-locale-filter-list="zh-hans" lang="zh-hans"><dfn>可本地化属性</dfn></span>
+        </h2>
+        <p its-locale-filter-list="en" lang="en">
+            The following attributes are localizable and share the same default configuration of <code>dir</code> and <code>lang</code> if not otherwise specified.
+        </p>
+        <p its-locale-filter-list="zh-hans" lang="zh-hans">
+            如下属性为可本地化属性。无特别规定的情况下，共享默认的<code>dir</code> and <code>lang</code>配置.
```
Comment (xfq) [inline specs/manifest/index.html:865]:
```suggestion
            如下属性为可本地化属性。无特别规定的情况下，共享默认的<code>dir</code>和<code>lang</code>配置。
```

File: specs/manifest/index.html [hunk 6]
```diff
@@ -846,6 +853,29 @@ <h2>
...
+        </p>
+        <ul>
+            <li>appName</li>
+            <li>shortName</li>
+            <li>description</li>
+            <li>window.navigationBarTitleText</li>
+            <li>widgets.name</li>
```
Comment (xfq) [inline specs/manifest/index.html:872]:
```suggestion
            <li><a href="#appname">appName</a></li>
            <li><a href="#shortname">shortName</a></li>
            <li><a href="#description">description</a></li>
            <li><a href="#window">window.navigationBarTitleText</a></li>
            <li><a href="#widgets">widgets.name</a></li>
```

Comment (xfq) [reply specs/manifest/index.html:872]: I think it would be useful to add some links here.

File: specs/manifest/index.html (hunk 1 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:314]: Thanks for spotting those.

File: specs/manifest/index.html (hunk 5 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:865]: Accepted.

File: specs/manifest/index.html (hunk 4 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:759]: Accepted.

File: specs/manifest/index.html (hunk 3 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:700]: Accepted.

File: specs/manifest/index.html (hunk 2 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:327]: Accepted.

File: specs/manifest/index.html (hunk 1 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:314]: Accepted.

File: specs/manifest/index.html (hunk 6 above)
Comment (zhangyongjing) [reply specs/manifest/index.html:872]: Accepted.

File: specs/manifest/index.html [hunk 7]
```diff
@@ -321,11 +321,10 @@ <h2>
           <span its-locale-filter-list="zh-hans" lang="zh-hans">语言</span>
         </h2>
         <p its-locale-filter-list="en" lang="en">
-            <code>lang</code> specifies the language tag of MiniApp. The value here takes the form of
-            <code>Language-Tag</code> as defined in the [[BCP47]]</a>. The default value is <code>zh-Hans</code>. If not specified, the default value is used.
+            <code>lang</code> specifies the default language of the MiniApp. It's applied to all <a>localizable attributes</a> of the manifest. The value here MUST be a well-formed <code>Language-Tag</code> specified in [[BCP47]].  The default value is undefined. Implementers are encouraged to provide the accurate value that matches the manifest content.
         </p>
         <p its-locale-filter-list="zh-hans" lang="zh-hans">
-            <code>lang</code>指定MiniApp的语言标签。这里取值为<code>Language-Tag</code>值，详见[[BCP47]]</a>。默认值是<code>zh-Hans</code>。未指定的情况下，将采用默认值。
+            <code>lang</code>定义了MiniApp的默认语言，适用于manifest所有<a>可本地化属性</a>，取值必须为[[BCP47]]所规定的合法的（well-formed）</well-formed><code>Language-Tag</code>值。默认值未定义。开发者需要提供匹配manifest内容的准确值。
```
Comment (xfq) [inline specs/manifest/index.html:327]:
```suggestion
            <code>lang</code>定义了MiniApp的默认语言，适用于manifest所有<a>可本地化属性</a>，取值必须为[[BCP47]]所规定的合法的（well-formed）<code>Language-Tag</code>值。默认值未定义。开发者需要提供匹配manifest内容的准确值。
```

Comment (xfq) [reply specs/manifest/index.html:327]: Looks like a typo here.

Comment (zhangyongjing) [reply specs/manifest/index.html:327]: indeed.

**Comment by zhangyongjing** [comment]
Can we close all related i18n issues accordingly? @xfq

**Comment by xfq** [comment]
I will comment in each issue.

**Pull request merged**